app/controllers/shf_applications_controller.rb

```diff
@@ -113,10 +113,24 @@ def update
         helpers.flash_message(:notice, t('.success'))
       end
 
-      redirect_to define_path(evaluate_update(params))
+      respond_to do |format|
```

Comment:
This should be a separate method and route.
This will only ever be an XHR request and will only ever be dealing with the uploaded files.  So it's not a match with everything that is going on in `update`. (no need to validate the company number(s) for example, etc etc.).
(We do this in CompaniesController, for example.)  

ex: `def delete_an_uploaded_file` 

The (separate) method will also make the intent very clear (e.g.  the only data involved will be the uploaded files, it must be xhr, is just doing deletion, ...)